packages/paste-core/components/corner-ornament/src/mappings.tsx

```diff
@@ -0,0 +1,69 @@
+import { BoxStyleProps } from "@twilio-paste/box";
+import { IconSizeOptions } from "@twilio-paste/style-props";
+
+import { CornerOrnamentPosition, CornerOrnamentType } from "./types";
+
+const BadgeBottomEndPath = "M1.015.6A.527.527 0 0 0 .5-.025a.525.525 0 1 0 .102 1.04A.367.367 0 0 1 .975.6h.04Z";
+const BadgeTopEndPath = "M.975.4H.958a.367.367 0 0 1-.356-.415A.528.528 0 0 0-.025.5a.525.525 0 1 0 1.04-.1h-.04Z";
+const IconTopEndPath =
+  "M.606-.014A.527.527 0 0 0-.025.5a.525.525 0 1 0 1.04-.106A.352.352 0 0 1 .6.05C.6.028.602.007.606-.014Z";
+const IconBottomEndPath =
+  "M1.014.606A.527.527 0 0 0 .5-.025a.525.525 0 1 0 .106 1.04A.352.352 0 0 1 .95.6c.022 0 .043.002.064.006Z";
+const AvatarBottomEndPath = "M1.023.546a.525.525 0 1 0-.477.477.35.35 0 0 1 .477-.477Z";
+const AvatarTopEndPath = "M1.023.454a.35.35 0 0 1-.477-.477.525.525 0 1 0 .477.477Z";
+
+export const ClipPathMappings: Record<CornerOrnamentType, Record<CornerOrnamentPosition, string>> = {
+  badge: {
+    bottom_end: BadgeBottomEndPath,
+    top_end: BadgeTopEndPath,
+  },
+
+  icon: {
+    bottom_end: IconBottomEndPath,
+    top_end: IconTopEndPath,
+  },
+  avatar: {
+    bottom_end: AvatarBottomEndPath,
+    top_end: AvatarTopEndPath,
+  },
+};
+
+export const OrnamentPositionStyleMappings: Record<
+  CornerOrnamentType,
+  Record<CornerOrnamentPosition, Partial<Record<IconSizeOptions, BoxStyleProps>>>
+> = {
+  badge: {
+    bottom_end: {
+      sizeIcon80: { left: "28px", top: "28px" },
+    },
```

Comment:
question: Can you explain how Corner Ornament works a bit?

Is it that _only_ sizeIcon80 can be used for the base component when Badge is used as the ornament? And then this specifies that the Badge should be placed on the bottom right, 28px from the left edge and 28px from the top edge?

Reply:
It's exactly that! By using this mapping we can control exactly what is supported (what they should and should not be using in combination). If they try and use another size with the badge for example they will get an error message that it is not supported and to raise it in discussions if it needed. It also allows us to be sure that ornaments and sizes align correctly across sizes.